indra/sources/medscan/api.py

```diff
@@ -1,13 +1,101 @@
 import codecs
 import logging
 import lxml.etree
-from collections import namedtuple
+from collections import namedtuple, defaultdict
 from indra.sources.medscan.processor import *
+import os
+import glob
+import tempfile
+import shutil
+from math import floor
+import pickle
+from indra.sources.medscan.fix_csxml_character_encoding import \
+        fix_character_encoding
+import time
 
 logger = logging.getLogger('medscan')
 
+def process_directory_statements_sorted_by_pmid(directory_name,
+                                                medscan_resource_dir):
+    """Processes a directory filled with CSXML files. For each file, first
```

Comment:
Docstring should start with single line describing what the function does/returns. More text can be provided after a blank line below that. The parameters and returns should also be documented.